copy/copy.go

```diff
@@ -357,9 +358,42 @@ func (c *copier) copy(ctx context.Context, src, srcComponents, target string, ov
 		}
 		if matchesExcludePattern {
 			include = false
+			excluded = true
+		}
+
+		// Optimization: Skip os.Lstat() for excluded paths when safe to do so.
+		// We can skip Lstat if:
+		// 1. The path is explicitly excluded
+		// 2. There are no include patterns (no need to check children)
+		// 3. There are no negation patterns in excludes (no exceptions to exclusions)
+		//
+		// This prevents "Access is denied" errors on Windows protected folders
+		// like "System Volume Information" and "WcSandboxState".
+		canSkip := !include && c.includePatternMatcher == nil &&
+			(c.excludePatternMatcher == nil || !c.excludePatternMatcher.Exclusions())
+		if canSkip {
+			return nil
 		}
 	}
 
+	if c.testHookLstat != nil {
+		c.testHookLstat(src)
+	}
+	fi, err := os.Lstat(src)
+	if err != nil {
+		return errors.Wrapf(err, "failed to stat %s", src)
```

Comment:
FWIW, `os.Lstat` (and many of the `os` functions) returns a `PathError`, which already includes the command executed and the path executed on;
https://github.com/golang/go/blob/2b62144069a130cc469f33009c0c392cc6de8810/src/os/stat_unix.go#L48


So wrapping it may make the error rather verbose (and duplicate the information); https://go.dev/play/p/pslhHfcWKDX

```console
failed to stat /no/such/file: lstat /no/such/file: no such file or directory
```

Reply:
Oh! I see that was already present, just moved (still worth looking at, perhaps for a follow-up 😅)